### ISSUE TITLE
Can't enter Debian or Ubuntu distrobox

### ISSUE THREAD

**Issue by pl43** [89luca89/distrobox#204]
Hi everyone! 

I'm having some issues with Debian related distrobox containers.

**Describe the bug**
Today my Debian Unstable distrobox stopped working correctly, as I couldn't update the system with apt. It turned out that the internet connection was completely gone, without modifying anything related to it. I removed the container (I've realised now, that it was a bad idea, I'm sorry :worried:) and created a new one, as I did earlier with:
`distrobox-create --name sid --image docker.io/library/debian:unstable --home /var/home/pt/Containers/sid/`

When I try to enter the container with `distrobox-enter sid` I get an error:

```
Starting container sid
run this command to follow along:
	podman logs -f sid
.......2022-03-16T18:07:57.688628000+01:00 /usr/bin/find
2022-03-16T18:07:57.689034000+01:00 /bin/mount
2022-03-16T18:07:57.689202000+01:00 /usr/bin/passwd
2022-03-16T18:07:57.689736000+01:00 /usr/bin/apt-get
2022-03-16T18:07:57.723017000+01:00 Ign:1 http://deb.debian.org/debian unstable InRelease
2022-03-16T18:07:58.724438000+01:00 Ign:1 http://deb.debian.org/debian unstable InRelease
2022-03-16T18:08:00.725512000+01:00 Ign:1 http://deb.debian.org/debian unstable InRelease
2022-03-16T18:08:04.726922000+01:00 Err:1 http://deb.debian.org/debian unstable InRelease
2022-03-16T18:08:04.727549000+01:00   Temporary failure resolving 'deb.debian.org'
2022-03-16T18:08:04.736240000+01:00 Reading package lists...
2022-03-16T18:08:04.752121000+01:00 Reading package lists...
2022-03-16T18:08:04.755278000+01:00 Building dependency tree...
2022-03-16T18:08:04.755537000+01:00 Reading state information...
2022-03-16T18:08:04.757445000+01:00 Error: An error occurred

An error occurred
```
The same thing happens when I try to enter a newly created Ubuntu 22.04 container:

```
distrobox-enter ubi
Starting container ubi
run this command to follow along:
	podman logs -f ubi
.......2022-03-16T18:10:24.773585000+01:00 /usr/bin/find
2022-03-16T18:10:24.773745000+01:00 /usr/bin/mount
2022-03-16T18:10:24.773898000+01:00 /usr/bin/passwd
2022-03-16T18:10:24.774232000+01:00 /usr/bin/apt-get
2022-03-16T18:10:24.799456000+01:00 Ign:1 http://archive.ubuntu.com/ubuntu jammy InRelease
2022-03-16T18:10:24.799742000+01:00 Ign:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
2022-03-16T18:10:24.799876000+01:00 Ign:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
2022-03-16T18:10:24.800217000+01:00 Ign:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
2022-03-16T18:10:25.802118000+01:00 Ign:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
2022-03-16T18:10:25.802879000+01:00 Ign:1 http://archive.ubuntu.com/ubuntu jammy InRelease
2022-03-16T18:10:25.803020000+01:00 Ign:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
2022-03-16T18:10:25.803669000+01:00 Ign:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
2022-03-16T18:10:27.804305000+01:00 Ign:1 http://archive.ubuntu.com/ubuntu jammy InRelease
2022-03-16T18:10:27.806731000+01:00 Ign:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
2022-03-16T18:10:27.806887000+01:00 Ign:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
2022-03-16T18:10:27.807007000+01:00 Ign:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
2022-03-16T18:10:31.804899000+01:00 Err:1 http://archive.ubuntu.com/ubuntu jammy InRelease
2022-03-16T18:10:31.805560000+01:00   Temporary failure resolving 'archive.ubuntu.com'
2022-03-16T18:10:31.805696000+01:00 Err:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
2022-03-16T18:10:31.805852000+01:00   Temporary failure resolving 'security.ubuntu.com'
2022-03-16T18:10:31.805976000+01:00 Err:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
2022-03-16T18:10:31.806094000+01:00   Temporary failure resolving 'archive.ubuntu.com'
2022-03-16T18:10:31.806856000+01:00 Err:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
2022-03-16T18:10:31.807046000+01:00   Temporary failure resolving 'archive.ubuntu.com'
2022-03-16T18:10:31.821859000+01:00 Reading package lists...
2022-03-16T18:10:31.845296000+01:00 Reading package lists...
2022-03-16T18:10:31.848720000+01:00 Building dependency tree...
2022-03-16T18:10:31.849006000+01:00 Reading state information...
2022-03-16T18:10:31.851976000+01:00 Error: An error occurred

An error occurred
```
It seems like the network connection is still not working. The problem is not present in my Fedora 35 container, everything is working as intended. I've even created a new Fedora container and it is working too. 

I attached the podman logs for both the Debian and Ubuntu containers.

--

**Logs**

```
podman logs sid
+ [ ! -f /run/.containerenv ]
+ [ -z 1000 ]
+ [ -z /var/home/pt/Containers/sid/ ]
+ [ -z pt ]
+ [ -z 1000 ]
+ basename /bin/bash
+ shell_pkg=bash
/usr/bin/find
+ command -v find
+ command -v mount
/bin/mount
/usr/bin/passwd
+ command -v passwd
+ command -v sudo
+ command -v apk
+ command -v apt-get
/usr/bin/apt-get
+ export DEBIAN_FRONTEND=noninteractive
+ apt-get update
Ign:1 http://deb.debian.org/debian unstable InRelease
Ign:1 http://deb.debian.org/debian unstable InRelease
Ign:1 http://deb.debian.org/debian unstable InRelease
Err:1 http://deb.debian.org/debian unstable InRelease
  Temporary failure resolving 'deb.debian.org'
Reading package lists...
W: Failed to fetch http://deb.debian.org/debian/dists/unstable/InRelease  Temporary failure resolving 'deb.debian.org'
W: Some index files failed to download. They have been ignored, or old ones used instead.
+ apt-get install -y bash findutils libnss-myhostname libvte-common ncurses-base passwd procps sudo util-linux
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libnss-myhostname
E: Unable to locate package libvte-common
E: Unable to locate package procps
E: Unable to locate package sudo
Error: An error occurred
+ [ 100 -ne 0 ]
+ printf Error: An error occurred\n
+ [ ! -f /run/.containerenv ]
+ [ -z 1000 ]
+ [ -z /var/home/pt/Containers/sid/ ]
+ [ -z pt ]
+ [ -z 1000 ]
+ basename /bin/bash
+ shell_pkg=bash
+ command -v find
/usr/bin/find
+ command -v mount
/bin/mount
/usr/bin/passwd
+ command -v passwd
+ command -v sudo
+ command -v apk
/usr/bin/apt-get
+ command -v apt-get
+ export DEBIAN_FRONTEND=noninteractive
+ apt-get update
Ign:1 http://deb.debian.org/debian unstable InRelease
Ign:1 http://deb.debian.org/debian unstable InRelease
Ign:1 http://deb.debian.org/debian unstable InRelease
Err:1 http://deb.debian.org/debian unstable InRelease
  Temporary failure resolving 'deb.debian.org'
Reading package lists...
W: Failed to fetch http://deb.debian.org/debian/dists/unstable/InRelease  Temporary failure resolving 'deb.debian.org'
W: Some index files failed to download. They have been ignored, or old ones used instead.
+ apt-get install -y bash findutils libnss-myhostname libvte-common ncurses-base passwd procps sudo util-linux
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libnss-myhostname
E: Unable to locate package libvte-common
E: Unable to locate package procps
E: Unable to locate package sudo
+ [ 100 -ne 0 ]
Error: An error occurred
+ printf Error: An error occurred\n
+ [ ! -f /run/.containerenv ]
+ [ -z 1000 ]
+ [ -z /var/home/pt/Containers/sid/ ]
+ [ -z pt ]
+ [ -z 1000 ]
+ basename /bin/bash
+ shell_pkg=bash
+ command -v find
/usr/bin/find
+ command -v mount
+ command -v passwd
/bin/mount
/usr/bin/passwd
+ command -v sudo
+ command -v apk
+ command -v apt-get
/usr/bin/apt-get
+ export DEBIAN_FRONTEND=noninteractive
+ apt-get update
Ign:1 http://deb.debian.org/debian unstable InRelease
Ign:1 http://deb.debian.org/debian unstable InRelease
Ign:1 http://deb.debian.org/debian unstable InRelease
Err:1 http://deb.debian.org/debian unstable InRelease
  Temporary failure resolving 'deb.debian.org'
Reading package lists...
W: Failed to fetch http://deb.debian.org/debian/dists/unstable/InRelease  Temporary failure resolving 'deb.debian.org'
W: Some index files failed to download. They have been ignored, or old ones used instead.
+ apt-get install -y bash findutils libnss-myhostname libvte-common ncurses-base passwd procps sudo util-linux
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libnss-myhostname
E: Unable to locate package libvte-common
E: Unable to locate package procps
E: Unable to locate package sudo
+ [ 100 -ne 0 ]
+ printf Error: An error occurred\n
Error: An error occurred
+ [ ! -f /run/.containerenv ]
+ [ -z 1000 ]
+ [ -z /var/home/pt/Containers/sid/ ]
+ [ -z pt ]
+ [ -z 1000 ]
+ basename /bin/bash
+ shell_pkg=bash
+ command -v find
/usr/bin/find
+ command -v mount
/bin/mount
+ command -v passwd
+ command -v sudo
/usr/bin/passwd
+ command -v apk
+ command -v apt-get
/usr/bin/apt-get
+ export DEBIAN_FRONTEND=noninteractive
+ apt-get update
Ign:1 http://deb.debian.org/debian unstable InRelease
Ign:1 http://deb.debian.org/debian unstable InRelease
Ign:1 http://deb.debian.org/debian unstable InRelease
Err:1 http://deb.debian.org/debian unstable InRelease
  Temporary failure resolving 'deb.debian.org'
Reading package lists...
W: Failed to fetch http://deb.debian.org/debian/dists/unstable/InRelease  Temporary failure resolving 'deb.debian.org'
W: Some index files failed to download. They have been ignored, or old ones used instead.
+ apt-get install -y bash findutils libnss-myhostname libvte-common ncurses-base passwd procps sudo util-linux
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libnss-myhostname
E: Unable to locate package libvte-common
E: Unable to locate package procps
E: Unable to locate package sudo
Error: An error occurred
+ [ 100 -ne 0 ]
+ printf Error: An error occurred\n
```
```
podman logs ubi
+ [ ! -f /run/.containerenv ]
+ [ -z 1000 ]
+ [ -z /var/home/pt/Containers/ubi/ ]
+ [ -z pt ]
+ [ -z 1000 ]
+ basename /bin/bash
+ shell_pkg=bash
+ command -v find
/usr/bin/find
+ command -v mount
/usr/bin/mount
+ command -v passwd
/usr/bin/passwd
+ command -v sudo
+ command -v apk
+ command -v apt-get
/usr/bin/apt-get
+ export DEBIAN_FRONTEND=noninteractive
+ apt-get update
Ign:1 http://archive.ubuntu.com/ubuntu jammy InRelease
Ign:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
Ign:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
Ign:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
Ign:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
Ign:1 http://archive.ubuntu.com/ubuntu jammy InRelease
Ign:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
Ign:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
Ign:1 http://archive.ubuntu.com/ubuntu jammy InRelease
Ign:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
Ign:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
Ign:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
Err:1 http://archive.ubuntu.com/ubuntu jammy InRelease
  Temporary failure resolving 'archive.ubuntu.com'
Err:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
  Temporary failure resolving 'security.ubuntu.com'
Err:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
  Temporary failure resolving 'archive.ubuntu.com'
Err:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
  Temporary failure resolving 'archive.ubuntu.com'
Reading package lists...
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease  Temporary failure resolving 'archive.ubuntu.com'
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/jammy-updates/InRelease  Temporary failure resolving 'archive.ubuntu.com'
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/jammy-backports/InRelease  Temporary failure resolving 'archive.ubuntu.com'
W: Failed to fetch http://security.ubuntu.com/ubuntu/dists/jammy-security/InRelease  Temporary failure resolving 'security.ubuntu.com'
W: Some index files failed to download. They have been ignored, or old ones used instead.
+ apt-get install -y bash findutils libnss-myhostname libvte-common ncurses-base passwd procps sudo util-linux
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libnss-myhostname
E: Unable to locate package libvte-common
E: Unable to locate package sudo
+ [ 100 -ne 0 ]
Error: An error occurred
+ printf Error: An error occurred\n
```

**Desktop (please complete the following information):**
 - OS: Fedora Silverblue
 - Version 35

I hope this issue is not related to the host system's network config. Everything is at default as the fedora installer configured it.

Anyway this software is brilliant and really helpful in my work, so thank you for maintaining it! :clap: I hope this bug report is helpful for you in some way.

All the best!

**Comment by 89luca89** [comment]
Hi @pl43 I've found there was an issue and fixed in this commit earlier: https://github.com/89luca89/distrobox/commit/333ae0ef6766ef50443dfd6735770e62014ac00c

Can you try using latest main version branch?

**Comment by meisme-dev** [comment]
The latest branch doesn't work for me, I'm on Fedora 35.

**Comment by jharmer95** [comment]
This seems to be affecting multiple distros, including Arch and Manjaro as well.

The patch you linked fixed it for me though. Thanks!

**Comment by fariszr** [comment]
I faced this issue as well.
Forcing distrobox to use docker instead of podman by removing it, fixed it.

**Comment by 89luca89** [comment]
Good, if the patch solved the issue I'll close this, if the problem persists feel free to reopen or open a new one :+1:

**Comment by fariszr** [comment]
Shouldn't this issue remain open until there is a stable release fixing it ?
Or at least it should be pinned.

**Comment by 89luca89** [comment]
Good point so we can redirect problems here :+1:

**Comment by pl43** [comment]
> Hi @pl43 I've found there was an issue and fixed in this commit earlier: [333ae0e](https://github.com/89luca89/distrobox/commit/333ae0ef6766ef50443dfd6735770e62014ac00c)
> 
> Can you try using latest main version branch?

Hi @89luca89! I installed the latest main version as you suggested, and everything is working now. Thank you very much for your help!

**Comment by aaylnx** [comment]
I'm having the same problem as others. I am running this on Fedora Silverblue. I am unsure how to apply the patch or to install using the different branch. Can anyone help? Thanks!

**Comment by 89luca89** [comment]
hi @aaylnx you can either git clone the repo and do a `./install` or download the sources zip [here](https://github.com/89luca89/distrobox/archive/refs/heads/main.zip), unzip and to `./install` from within the unzipped directory

**Comment by shibumi** [comment]
Hi, can we get a new release with this patch :) people on Fedora are running into this issue

**Comment by misobarisic** [comment]
> Hi, can we get a new release with this patch :) people on Fedora are running into this issue

I suggest installing the latest `main` commit manually until a new version of distrobox is released.

**Comment by 89luca89** [comment]
Fixed in release 1.2.14
